### PR TITLE
Require Node.js 6, add TypeScript definition, update dependencies

### PR DESCRIPTION
**Pull request by BendingBender**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	Result1,
+	Result2,
+	Result3,
+	Result4,
+	Result5,
+	Result6,
+	Result7,
+	Result8,
+	Result9,
+	Result10
```
Comment (sindresorhus) [inline index.d.ts]: What if there are more than 10 inputs? There must be a better way of achieving this...

File: index.test-d.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	() => Promise<void>,
+	() => Promise<number>
+] = [
+	() => Promise.resolve('sindresorhus.com'),
+	() => Promise.resolve('ava.li'),
+	() => Promise.resolve(),
+	() => Promise.resolve(1),
+];
+
+pAll(actions, { concurrency: 2 }).then(result => {
```
Comment (sindresorhus) [inline index.test-d.ts]: Use async/await.

File: index.d.ts (hunk 1 above)
Comment (BendingBender) [reply index.d.ts]: There is a not-so-convenient fallback for any number of parameters:
```ts
declare function pAll<TAll>(
	tasks: Iterable<PromiseFactory<TAll>> | PromiseFactory<TAll>[],
	options?: Options
): Promise<TAll[]>;
```

ATM, TypeScript doesn't have a better way to describe this one.

Comment (sindresorhus) [reply index.d.ts]: Do you know if there's any TS issue for this? Seems like a very big limitation.

Comment (privatenumber) [reply index.d.ts:28]: Seems like you've already found it, but for anyone else that's reading this there's a related issue in https://github.com/microsoft/TypeScript/issues/5453, which was addressed via variadic tuple types, but unfortunately [doesn't solve this yet](https://github.com/microsoft/TypeScript/issues/39788)

**Pull request merged**